code/modules/projectiles/updated_projectiles/guns/specialist.dm

```diff
@@ -799,52 +809,26 @@
 		current_mag.current_rounds++
 	return TRUE
 
-
-/obj/item/weapon/gun/launcher/rocket/reload(mob/user, obj/item/ammo_magazine/rocket)
-	if(flags_gun_features & GUN_BURST_FIRING)
+/obj/item/weapon/gun/launcher/rocket/unload(mob/user)
+	if(!user)
 		return
-
-	if(!rocket || !istype(rocket) || rocket.caliber != current_mag.caliber)
-		to_chat(user, "<span class='warning'>That's not going to fit!</span>")
+	if(!current_mag || isnull(current_mag) || current_mag.loc != src)
```

Comment:
```suggestion
	if(!current_mag || current_mag.loc != src)
```
`isnull(A)` will always be false if `!A` is false.

Reply:
This is how it's set up in the reload proc; I assume it's this way for a reason